src/BatteryStateDisplayService.cpp

```diff
@@ -27,7 +27,16 @@ void BatteryStateDisplayService::handleBatteryDataReceived(const BatteryData& ba
 
     QTextStream(stdout) << "Voltage: " << batteryData.voltage
         << " Current: " << batteryData.current
-        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed() << endl;
-
+        << " Total Ah used: " << batteryStateOfChargeService_.totalAmpHoursUsed()<<endl;
     // TODO Print out time till it is depleted or charged.
+    if (batteryStateOfChargeService_.isCharging())
+        QTextStream(stdout)<<"Time left till full charge: "
+        <<batteryStateOfChargeService_.timeWhenChargedOrDepleted().toString("hh:mm:ss.zzz")<<endl;
```

Comment:
For string and numeric constants, it is a good programming practice to extract them out into named constants. In c++ one way of doing it is to use anonymous namespaces near the top of the file. The syntax would be like:

```c++
namespace
{
   const QString TIME_FORMAT = "hh:mm:ss.zzz";
}
```

The benefit for doing it this way is that if the format ever changes then you don't need to change it in multiple places.